lib/mayaUsd/ufe/UsdAttributes.cpp

```diff
@@ -311,6 +304,37 @@ bool UsdAttributes::hasAttribute(const std::string& name) const
     return false;
 }
 
+#if (UFE_PREVIEW_VERSION_NUM >= 4024)
+Ufe::Attribute::Ptr
+UsdAttributes::addAttribute(const std::string& name, const Ufe::Attribute::Type& type)
```

Comment:
Really feels like if the base UFE interface had a default implementation for this that created the command and called execute on it, you wouldn't be doing too much additional work.  What do you think?  I assume what you have is slightly higher performance, because there is no state delegate keeping track of USD changes, but I wonder if that really makes any practical difference.